### ISSUE TITLE
Crash when running autoComplete/values for tags with space

### ISSUE THREAD

**Issue by Hipska** [go-graphite/carbonapi#420]
When using [graphite-clickhouse](https://github.com/lomik/graphite-clickhouse) as backend, it is possible to have spaces in tag values. But when using or searching for them with Grafana, it occurs that `carbonapi` crashes. 

Here is the stack trace:
```log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xcc4710]
goroutine 82 [running]:
github.com/go-graphite/carbonapi/zipper/errors.(*Errors).Add(...)
/root/go/src/github.com/go-graphite/carbonapi/zipper/errors/errors.go:75
github.com/go-graphite/carbonapi/zipper/protocols/v2.(*ClientProtoV2Group).doTagQuery(0xc000169200, 0x100dd20, 0xc0000b8f80, 0x0, 0xc00009c700, 0xb2, 0x2710, 0x0, 0x0, 0x0, ...)
/root/go/src/github.com/go-graphite/carbonapi/zipper/protocols/v2/protobuf_group.go:351 +0x2d0
github.com/go-graphite/carbonapi/zipper/protocols/v2.(*ClientProtoV2Group).TagValues(0xc000169200, 0x100dd20, 0xc0000b8f80, 0xc00009c700, 0xb2, 0x2710, 0x15, 0xc0002f07c8, 0xbdd4fb, 0x18b78c8)
/root/go/src/github.com/go-graphite/carbonapi/zipper/protocols/v2/protobuf_group.go:367 +0x6f
github.com/go-graphite/carbonapi/zipper/broadcast.(*BroadcastGroup).doTagRequest(0xc00017d320, 0x100dd20, 0xc0000b8f80, 0xc0000b7c80, 0x101ec00, 0xc000169200, 0xe0a440, 0xc0000a9260, 0xc0000b7b60)
/root/go/src/github.com/go-graphite/carbonapi/zipper/broadcast/broadcast_group.go:463 +0x6f5
created by github.com/go-graphite/carbonapi/zipper/types.DoRequest
/root/go/src/github.com/go-graphite/carbonapi/zipper/types/response.go:43 +0x224
```

If you don't want to support tag values having spaces, then please just make sure that carbonapi does not crash, but instead gives an empty result?

**Comment by Civil** [comment]
A quick question - you are running current master or the latest release? To my knowledge that issue should be fixed in master.

**Comment by robertely** [comment]
I'm seeing the same error in the current release.

 I don't have a tagdb configured, i was wondering if that had something to do with it.

**Comment by Hipska** [comment]
I use latest version from https://packagecloud.io/go-graphite which seems to be 0.12.5

**Comment by Civil** [comment]
Could both of you try latest build available?

https://packagecloud.io/go-graphite/autobuilds for example here the latest one, which is carbonapi_0.12.5-4-g6a93c9 ?

As this issue likely to be fixed by https://github.com/go-graphite/carbonapi/commit/d0290dafd76ab2d93d82ef83bf5df31c647ec8b2 commit, but I haven't tagged a release since then as I want to do a bigger one with reworked error handling logic.

**Comment by robertely** [comment]
Absolutely, i'm working on it now. 

Autobuilds don't seem to build RPMs so i'm compiling against master.

**Comment by robertely** [comment]
It looks like  [d0290da](https://github.com/go-graphite/carbonapi/commit/d0290dafd76ab2d93d82ef83bf5df31c647ec8b2) fixed it. I'm testing it a little more exhaustively now, but it's working. 

Btw: My work around is to run in graphite 9.x mode as set in the config and grafana. Not perfect but it generally keeps grafana from querying for tags. 

Setting it to the 1.x protocol leads to an almost instant crash when you go anywhere near the find api. It's how I'm replicating it in testing.

**Comment by Hipska** [comment]
Same here, also running on RH, so an RPM with a minor version bump would be nice.

**Comment by Civil** [comment]
Fixed in 0.12.6

**Comment by ghmeier** [comment]
Will ya'll be making 0.12.6 stable since the current stable build (0.12.5) has this error?
